### ISSUE TITLE
Wrong calculation of tensor inverse?

### ISSUE THREAD

**Issue by Peter9606** [NVIDIA/cutlass#182]
https://github.com/NVIDIA/cutlass/blob/ccb697bac77fcc898e9c897b2c90aa5b60ac72fb/include/cutlass/layout/tensor.h#L152-L156

I'm thinking if the above lines should be corrected to

```
    n = int(index / stride_[2]);
    LongIndex residual = index % stride_[2];
    h = int(residual / stride_[1]);`
    residual = (residual % stride_[1]);
```

**Comment by hwu36** [comment]
I think you are right.  We will fix this.

Thank you very much!

**Comment by Peter9606** [comment]
There is a unit test case against this method named `Layout_TensorNHWC`, but probably it is insufficient or incorrect because it will get passed even force `inverse` fall into the wrong path.

**Comment by hwu36** [comment]
The unit test is insufficient to test this.

For example, a 2x16x8x32 tensor is treated as 1x1x256x1 tensor by mistake.  Then the input coordinate [1, 15, 7, 0] whose offset is 8160 is recognize as [0, 0, 255, 0] after inverse() in the new coordinate system whose offset is also 8160.

**Comment by d-k-b** [comment]
Fixed in 2.5.
